main.tf

```diff
@@ -1,4 +1,6 @@
 resource "aws_cloudfront_distribution" "cloudfront_distribution" {
+  count = var.create_cf ? 1 : 0
+
```

Comment:
Interesting idea.  presumably someone would only use this module if they actually want to deploy the resource, but I could see value in someone already having this in their config and wanting to toggle it on/off

Any specific reason there is an additional blank row?

Reply:
i'll fix the blank line no reason.
about the value of this, if you use this module as a nested modules, you can have multiple configuration in some you want to have the distribution and in some not, depending on your overall config

Reply:
fixed